src/ows/src/main/java/org/geoserver/ows/FilePublisher.java

```diff
@@ -72,7 +72,7 @@ protected void initServletContext(ServletContext servletContext) {
     @Override
     protected String getMimeType(String reqPath, String filename) {
         String mimeType = super.getMimeType(reqPath, filename);
-        if (!"/index.html".equals(reqPath)) {
+        if (!("/index.html".equals(reqPath) || "/accessDenied.html".equals(reqPath))) {
```

Comment:
Hey @sikeoka geoserver has maintained a reference to a long dead `accessDenied.jsp` for ages. Changing this to `accessDenied.html` and providing a mapping works but I had to look at your ``DISABLE_STATIC_WEB_FILES`` special case here which forces ``text/plain``.